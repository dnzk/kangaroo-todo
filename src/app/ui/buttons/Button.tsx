import style from "./buttons.module.scss"
import font from "@/app/lib/font"
import { DetailedHTMLProps, ButtonHTMLAttributes } from "react"

type Variant = "primary" | "secondary"

type ButtonProps = DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement> & { variant?: Variant }

export default function Button(props: ButtonProps) {
  const buttonVariant = props.variant || 'primary'
  const buttonStyle = buttonVariant === 'primary' ? style.primary : style.secondary

  return (
    <button className={`${font.className} ${buttonStyle}`} {...props}>{props.children}</button>
  )
}
