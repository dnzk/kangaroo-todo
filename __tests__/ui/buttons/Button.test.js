import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import Button from '../../../src/app/ui/buttons/Button'

describe('Button', () => {
  it('renders correctly', () => {
    const { container } = render(<Button />)
    expect(container).toMatchSnapshot()
  })
})
