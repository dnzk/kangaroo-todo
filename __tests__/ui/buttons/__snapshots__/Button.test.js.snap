// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<div>
  <button
    class="className primary"
  />
</div>
`;
